tinygrad/tensor.py

```diff
@@ -245,13 +243,17 @@ def register(name, fxn, gpu=False):
   else:
     Tensor.ops[name] = fxn
   def dispatch(*x, **kwargs):
+    x = tuple((Tensor(arg,requires_grad=False) if not isinstance(arg, Tensor) else arg for arg in x ))
     f = (Tensor.opsgpu if x[0].gpu else Tensor.ops)[name]
     f.cl_ctx, f.cl_queue = cl_ctx, cl_queue
     return f.apply(f, *x, **kwargs)
   setattr(Tensor, name, dispatch)
   if name in ['add', 'sub', 'mul', 'div', 'pow']:
-    setattr(Tensor, "__%s__" % name, dispatch)
-    setattr(Tensor, "__i%s__" % name, lambda self,x: self.assign(dispatch(self,x)))
+    setattr(Tensor, f"__{name}__", dispatch)
+    setattr(Tensor, f"__i{name}__", lambda self,x: self.assign(dispatch(self,x)))
+    if "__r%s__" % name in dir(int):
```

Comment:
Which does this fail for?

Reply:
"__rdiv__"

Reply:
Hmm, I think that's a different problem.

https://stackoverflow.com/questions/34344025/have-the-rdiv-and-idiv-operators-changed-in-python-3-x

Reply:
yeah we dont support right div in here